packages/app-builder-lib/src/targets/nsis/NsisTarget.ts

```diff
@@ -207,6 +207,10 @@ export class NsisTarget extends Target {
       APP_GUID: guid,
       // Windows bug - entry in Software\Microsoft\Windows\CurrentVersion\Uninstall cannot have \ symbols (dir)
       UNINSTALL_APP_KEY: uninstallAppKey,
+      UNINSTALL_URL_HELP: options.uninstallUrlHelp || this.packager.info.metadata.homepage || undefined,
+      UNINSTALL_URL_INFO_ABOUT: options.uninstallUrlInfoAbout || this.packager.info.metadata.homepage || undefined,
+      UNINSTALL_URL_UPDATE_INFO: options.uninstallUrlUpdateInfo || this.packager.info.metadata.homepage || undefined,
+      UNINSTALL_URL_README: options.uninstallUrlReadme || this.packager.info.metadata.homepage || undefined,
```

Comment:
I think `undefined` may have unexpected side effects. Note: `defines.ENABLE_LOGGING_ELECTRON_BUILDER = null` where `null` is used as a flag (I guess implicitly represents a boolean in the nsis scripts?)

Easy way around this - only set the defines var when provided a value
```
const { homepage } = this.packager.info.metadata
// `use` from builder-util IIRC
use(options.uninstallUrlHelp || homepage, (it) => defines.UNINSTALL_URL_HELP = it)
use(options.uninstallUrlInfoAbout || homepage, (it) => defines.UNINSTALL_URL_INFO_ABOUT = it)
use(options.uninstallUrlUpdateInfo || homepage, (it) => defines.UNINSTALL_URL_UPDATE_INFO = it)
use(options.uninstallUrlReadme || homepage, (it) => defines.UNINSTALL_URL_README = it)
```

Reply:
Also, wondering if we need to allow disabling the field from being set if the developer doesn't want their homepage to show there. Curious on your thoughts? Basically, `uninstallUrlHelp?: string | false`

Reply:
I don't think that's necessary. The only use case for defining the homepage is to show it somewhere, whether that's the npm registry or somewhere else, you want that homepage to be linked to the project.